m2cgen/assemblers/__init__.py

```diff
@@ -83,4 +86,10 @@ def get_assembler_cls(model):
         raise NotImplementedError(
             "Model {} is not supported".format(model_name))
 
+    if assembler_cls is XGBoostModelAssembler:
+        model_dump = model.get_booster().get_dump(dump_format="json")
+        if len(model_dump) == 1 and all(i in json.loads(model_dump[0])
```

Comment:
Looks like a logic that should reside within the assembler itself.

Reply:
Can you please elaborate a little bit more? Do you mean that the new class `XGBoostLinearModelAssembler` is not needed?

Reply:
I mean that we should have an umbrella `XGBoostModelAssembler` which will choose between the linear and tree version based on this logic. So I suggest 3 entities: `XGBoostTreeModelAssembler` - what is currently named `XGBoostModelAssembler`, `XGBoostLinearModelAssembler` - the implementation of which you've introduced and `XGBoostModelAssembler` - which contains the logic of choosing between `*Tree*` and `*Linear*`. 
What do you think?

Reply:
That sounds reasonable!